ntp-proto/src/time_types.rs

```diff
@@ -179,12 +179,34 @@ impl NtpDuration {
         Self { duration }
     }
 
+    pub(crate) fn from_exponent(input: i8) -> Self {
+        Self::from_seconds(log2d(input))
+    }
```

Comment:
so technically the exponents are `i8`s. In practice they are usually `u8`s so we may want a custom function for that (which could be `const`)